server/controllers/admin/placementRequests/withdrawalsController.ts

```diff
@@ -0,0 +1,52 @@
+import type { Request, RequestHandler, Response } from 'express'
+
+import { catchValidationErrorOrPropogate, fetchErrorsAndUserInput } from '../../../utils/validation'
+import paths from '../../../paths/admin'
+import { PlacementRequestService } from '../../../services'
+import { ErrorWithData } from '../../../utils/errors'
+
+export const tasklistPageHeading = 'Apply for an Approved Premises (AP) placement'
+
+export default class WithdrawlsController {
+  constructor(private readonly placementRequestService: PlacementRequestService) {}
+
+  new(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      const { errors, errorSummary } = fetchErrorsAndUserInput(req)
+
+      return res.render('admin/placementRequests/withdrawals/new', {
+        pageHeading: 'Are you sure you want to withdraw this placement request?',
+        id: req.params.id,
+        errors,
+        errorSummary,
+      })
+    }
+  }
+
+  create(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      try {
+        if (!req.body.confirm) {
+          throw new ErrorWithData({ 'invalid-params': [{ propertyName: `$.confirm`, errorType: 'empty' }] })
```

Comment:
Might be worth stealing/adapting `insertGenericError` from CAS3 - the API error formatting feels like something controllers shouldn't need to know about